Libraries/Image/RCTImageLoader.m

```diff
@@ -390,31 +401,6 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithURLRequest:(NSURLRequest *
       request = mutableRequest;
     }
 
-    // Check for cached response before reloading
-    // TODO: move URL cache out of RCTImageLoader into its own module
-    NSCachedURLResponse *cachedResponse = [_URLCache cachedResponseForRequest:request];
-
-    while (cachedResponse) {
-      if ([cachedResponse.response isKindOfClass:[NSHTTPURLResponse class]]) {
-        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)cachedResponse.response;
-        if (httpResponse.statusCode == 301 || httpResponse.statusCode == 302) {
```

Comment:
I tested that situation, and figured out that ```NSURLRequest(imageURLRequest)```'s URL wasn't same with ```NSHTTPURLResponse```'s URL(set as final location of file), and the response's status code is 200(or 404 if there are no file). I think without implementation about ```URLSession:willPerformHTTPRedirection```, the user-agent(in this case, iOS) try redirect and get data with final server's status code.